tests/testthat/setup-testing_depth.R

```diff
@@ -0,0 +1,56 @@
+#' Returns testing depth set by an environmental variable.
+#'
+#' @details Looks for the option `TESTING_DEPTH` first, if not set,
+#' takes the system environmental variable `TESTING_DEPTH`. If neither
+#' is set, then returns 3 by default. If the value of `TESTING_DEPTH`
+#' is not a scalar numeric, then returns 3.
+#'
+#' @return `numeric(1)` the testing depth.
+#'
+testing_depth <- function() {
+  testing_depth <- getOption("TESTING_DEPTH")
+  if (is.null(testing_depth)) {
```

Comment:
```suggestion
  if (is.null(testing_depth) & Sys.getenv("TESTING_DEPTH") != "") {
```

Reply:
I think the current logic is straightforward. We set depth from options, then set options from env.

Perhaps setting the `default` and `unset` to `3` for the `getOption` and `Sys.getenv` respectively?

Reply:
And keep this 3 as a default variable in the function as it's used in many places/

Reply:
What is not straightforward is that empty `getOption` return NULL and empty `Sys.getenv` returns `""` and then you need to do `tryCatch` on `as.numeric` and figure out if you got `NA`

Reply:
Now the empty `getOption` will return 3 and empty `Sys.getenv` will return "3"
84568ec81c453e3cb3a37b161661e3a3135f92ab